theano/tensor/nnet/abstract_conv.py

```diff
@@ -1517,18 +1525,23 @@ def conv(self, img, kern, mode="valid", dilation=1):
                      ] = kern
 
         if self.convdim == 2:
+            if img.shape[1] % self.num_groups != 0:
+                raise ValueError("Number of input channels must be divisible by num_groups")
             val = _valfrommode(mode)
             bval = _bvalfromboundary('fill')
+            input_channel_offset = img.shape[1] // self.num_groups
+            output_channel_offset = kern.shape[0]
```

Comment:
I did informally test this implementation by taking some input and kernel and splitting them into separate groups and then using normal convolutions. Then i concatenated the outputs and checked them against the outputs i got in grouped convolutions by giving the same input and kernel.  The difference was 1e-15 or less.
I checked the same for the backward passes as well.

yes,  line 1539 doesnt change behavior. I thought since the same value was being fetched, substituting them in both places would make them more readable. I was mistaken there.